protocols/mdns/src/behaviour/iface/query.rs

```diff
@@ -179,6 +179,7 @@ impl MdnsResponse {
     ) -> impl Iterator<Item = (PeerId, Multiaddr, Instant)> + '_ {
         self.discovered_peers()
             .filter(move |peer| peer.id() != &local_peer_id)
+            .filter(move |peer| now.checked_add(peer.ttl()).is_some())
```

Comment:
```suggestion
            .filter(move |peer| peer.id() != &local_peer_id && peer.ttl() > now)
```
No need for multiple `Filter` wrappers.

Reply:
Sorry, just saw that my above suggestion doesn't make any sense.